pygmt/datasets/earth_relief.py

```diff
@@ -36,7 +42,21 @@ def load_earth_relief(resolution="01d"):
 
     """
     _is_valid_resolution(resolution)
-    fname = which("@earth_relief_{}".format(resolution), download="u")
+
+    if registration in ("pixel", "gridline", None):
+        _registration = ""  # If None, let GMT decide on Pixel/Gridline type
+        with clib.Session() as lib:
+            if registration is not None and Version(lib.info["version"]) >= Version(
+                "6.1.0"
+            ):
+                _registration = f"_{registration[0]}"
```

Comment:
If GMT<6.1.0, we should raise an warning like "The 'registration' argument is only available for GMT>=6.1.0".

Reply:
`registration="gridline"` will work on GMT 6.0, so I'll just raise `GMTInvalidInput` when GMT 6.0 users try to pass in `registration="pixel"`.